### ISSUE TITLE
InnerBlock template outputs HTML even when nothing in the template is filled in

### ISSUE THREAD

**Issue by ghost** [WordPress/gutenberg#47956]
### Description

I used npx @wordpress/create-block to create a simple block and added some InnerBlocks with template and allowedBlocks like so:

edit.js
```
export default function Edit() {
	const ALLOWED_BLOCKS = ['core/group', 'core/image', 'core/paragraph', 'core/heading'];
	const MY_TEMPLATE = [
		['core/group', { 'lock': { 'move': true, 'remove': false } }, [
			['core/heading', { placeholder: 'Book Title' }],
			['core/paragraph', { placeholder: 'Summary' }],
		]],
		['core/group', { 'lock': { 'move': true, 'remove': false } }, [
			['core/image', {}],
		]],
	];

	return (
		<div { ...useBlockProps() }>
			<InnerBlocks
				allowedBlocks={ ALLOWED_BLOCKS }
				template={ MY_TEMPLATE }
				templateLock="all"
			/>
		</div>
	);
}
```

save.js
```
export default function save() {
	return (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	);
}
```

When I then only fill in the core/heading of the block template, all the other HTML of the block still gets rendered, polluting the HTML of the document. I expect only the filled out parts of the block template to be rendered.

Current output:
```
<div class="wp-block-create-block-static-block">
    <div class="wp-block-group is-layout-flow">
        <h2 class="wp-block-heading">Headline only!</h2>
        <p></p>
    </div>
    <div class="wp-block-group is-layout-flow">
        <figure class="wp-block-image">
            <img alt="">
        </figure>
    </div>
</div>
```

Expected output:
```
<div class="wp-block-create-block-static-block">
    <div class="wp-block-group is-layout-flow">
        <h2 class="wp-block-heading">Headline only!</h2>
    </div>
</div>
```

Basically everything that has no content will not be rendered. So also if the inside of a group is empty it will not be rendered.

### Step-by-step reproduction instructions

1. Go to the Block and only fill in parts of the block template
2. Look at the HTML in the frontend

### Screenshots, screen recording, code snippet
<img width="1086" alt="Screenshot 2023-02-10 at 09 20 23" src="https://user-images.githubusercontent.com/97944141/218040253-9d77c91b-e49b-4ed6-8332-fac303457cb1.png">

### Environment info

Gutenberg: Version 15.1.0
WordPress 6.1.1 
Theme: 2023

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
